lib/howl/commands/app_commands.moon

```diff
@@ -302,36 +313,54 @@ command.register
 -----------------------------------------------------------------------
 
 launch_cmd = (working_directory, cmd) ->
-  shell = howl.sys.env.SHELL or '/bin/sh'
-  p = Process {
-    :cmd,
-    :shell,
-    read_stdout: true,
-    read_stderr: true,
-    working_directory: working_directory,
-  }
-
-  buffer = ProcessBuffer p
-  editor = app\add_buffer buffer
-  editor.cursor\eof!
-  buffer\pump!
+  dispatch.launch ->
```

Comment:
Right. The handler is called in the same coroutine that called  `command.run` - so running a long running process doesn't work well because the command stays active and the command line doesn't close. (Maybe you remember the bug you reported..).